packages/node-core/src/indexer/project.service.spec.ts

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020-2023 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import {NodeConfig} from '../configure';
+import {BaseDsProcessorService} from './ds-processor.service';
+import {DynamicDsService} from './dynamic-ds.service';
+import {BaseProjectService} from './project.service';
+import {ISubqueryProject} from './types';
+
+class TestProjectService extends BaseProjectService<any, any> {
+  packageVersion = '1.0.0';
+
+  async getBlockTimestamp(height: number): Promise<Date> {
+    return Promise.resolve(new Date());
+  }
+
+  onProjectChange(project: any): void {
+    return;
+  }
+}
+
+describe('BaseProjectService', () => {
+  let service: TestProjectService;
+
+  beforeEach(() => {
+    service = new TestProjectService(
+      null as unknown as BaseDsProcessorService,
+      null as unknown as any,
+      null as unknown as any,
+      null as unknown as any,
+      null as unknown as any,
+      {dataSources: []} as unknown as ISubqueryProject<any>,
+      null as unknown as any,
+      null as unknown as any,
+      {unsafe: false} as unknown as NodeConfig,
+      {getDynamicDatasources: jest.fn()} as unknown as DynamicDsService<any>,
+      null as unknown as any,
+      null as unknown as any
+    );
+  });
+
+  test('hasDataSourcesAfterHeight', async () => {
```

Comment:
Please follow this pattern

```suggestion
  it('has data sources after height', async () => {
```